File: codingTest/src/codingTest/network.java
package codingTest;

public class network {

	public static void main(String[] args) {
		Solution s = new Solution();
		int result = s.solution(3, new int[][] { {1,1,0}, {1,1,0}, {0,0,1}});
		System.out.println("결과 : " + result);
	
	}
	
	public static class Solution {
		public int solution(int n, int[][] computers) {
			int answer = 0;
			boolean visited[] = new boolean[n];
			
			for(int i=0; i < n; i++) {
				if(!visited[i]) { // 방문하지 않은 노드만
					dfs(i, computers, visited); // 깊이탐색 실행
					answer++;	
				}
			}
			return answer;
		}
		
		public void dfs(int n, int[][] computers, boolean[] visited) {
			visited[n] = true; // 해당 노드를 방문한 것으로 변경
			
			for(int i=0; i < computers.length; i++) { // 해당 배열의 갯수만큼 반복
				// 방문하지 않았고, 해당 노드가 연결되어 있으면
				if(!visited[i] && computers[n][i] == 1) {
					dfs(i, computers, visited); // 그 노드로 이동해서 dfs 실행 (네트워크가 연결되어 있는지 확인하기 위해)
				}
			}
		}
	}
}

File: codingTest/src/codingTest/network.js
function solution (n, computers) {
    var answer = 0;
    var isVisited = new Boolean[n]

    var DFS = function(computers, i) {
        
        if(isVisited[i]) {return} //이미 방문했으면 return
        
        //방문했음
        isVisited[i] = true

        for(let j = 0; j < computers.length; j++) {
            if(computers[i][j] == 1) {
                DFS(computers, j)
            }
        }
    }


    for(let i = 0; i < n; i++){
        if(!isVisited[i]) {
            //방문 안했으면 탐색 !!
            answer++
            DFS(computers, i)
        }
    }
}